service/matching/matcher_test.go

```diff
@@ -140,37 +140,39 @@ func (t *MatcherTestSuite) TestRemoteSyncMatchBlocking() {
 
 func (t *MatcherTestSuite) testRemoteSyncMatch(taskSource enumsspb.TaskSource) {
 	pollSigC := make(chan struct{})
+	<-t.fwdr.PollReqTokenC()
 
 	go func() {
 		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
-		<-pollSigC
+		<-pollSigC // blocking call since we are receiving something
 		if taskSource == enumsspb.TASK_SOURCE_DB_BACKLOG {
 			// when task is from dbBacklog, sync match SHOULD block
 			// so lets delay polling by a bit to verify that
 			time.Sleep(time.Millisecond * 10)
 		}
-		task, err := t.childMatcher.Poll(ctx, &pollMetadata{})
+		task, err := t.rootMatcher.Poll(ctx, &pollMetadata{})
 		cancel()
+		t.False(task.isStarted())
```

Comment:
This was modified by-mistake. A follow up PR was pushed which reverted this back. Sorry!